File: my-folder/0268-missing-number/solution.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n=nums.size();
        int b=n;
        int sum1=0;
        while(b>0){
            sum1+=b;
            b--;
        }
        int sum2=0;
        for(int i=0;i<n;i++){
            sum2+=nums[i];
            }
        return sum1-sum2;
        }
    };

File: my-folder/0268-missing-number/solution.java
class Solution {
    public int missingNumber(int[] nums) {
        // int sum1=0;
        // int sum2=0;
        // int n=nums.length;
        // for(int i=0;i<n;i++){
        //     sum1+=nums[i];
        // }
        // while(n>0){
        //     sum2+=n;
        //     n--;
        // }
        // int sum=sum2-sum1;
        // return sum;
        int n=nums.length;
        int xor=0;
        for(int i=0;i<n;i++){
            xor=xor^i;
            xor=xor^nums[i];
        }
        xor=xor^n;
        return xor;
    }
}
